sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -634,13 +634,20 @@ def append_taskrun_params(task_name_append: str):
                         dst = '$(results.%s.path)' % sanitize_k8s_name(result['name'])
                         if artifact_name == result['name'] and src != dst:
                             add_copy_results_artifacts_step = True
+                            total_size_command = 'TOTAL_SIZE=$( expr $TOTAL_SIZE + $ARTIFACT_SIZE)\n'
+                            copy_command = '    cp ' + src + ' ' + dst + '\n'
+                            if env.get('OUTPUT_PREVIEW', 'false').lower() == 'true':
+                                preview_size = env.get('OUTPUT_PREVIEW_SIZE', '100')
+                                total_size_command = 'TOTAL_SIZE=$( expr $TOTAL_SIZE + %s)\n' % preview_size
+                                copy_command = '    dd if=' + src + ' of=' + \
+                                               dst + ' bs=' + preview_size + ' count=1\n'
                             script += (
                                     'ARTIFACT_SIZE=`wc -c %s | awk \'{print $1}\'`\n' % src +
-                                    'TOTAL_SIZE=$( expr $TOTAL_SIZE + $ARTIFACT_SIZE)\n' +
+                                    total_size_command +
                                     'touch ' + dst + '\n' +  # create an empty file by default.
                                     'if [[ $TOTAL_SIZE -lt 3072 ]]; then\n' +
                                     '  if ! awk "/[^[:print:]]/{f=1} END{exit !f}" %s; then\n' % src +
```

Comment:
Wouldn't it be more efficient to first copy the N bytes locally and then check that and if it matches, copy them to the destination?

Reply:
Maybe there's a better way to check whether the file is a binary. The original purpose for this condition is to not include any special character that could crash the Tekton and Kubernetes API server due to regular express parsing errors.

@yhwang do you happen to know any other command that can do it faster without print?

Reply:
may i can address this in another PR since this is not related to this feature.